MeesGame/Gameplay/Generator/MazeGenerator.cs

```diff
@@ -31,15 +72,29 @@ public static TileField GenerateMaze(int numRows = DEFAULT_NUM_ROWS, int numCols
                 }
             }
 
-            Random random = GameEnvironment.Random;
+            //Random random = GameEnvironment.Random;
             IList<Point> nodesToDo = new List<Point>();
             Point start = new Point(random.Next(tiles.Columns / 2 - 1) * 2 + 1, random.Next(tiles.Rows / 2 - 1) * 2 + 1);
+            bestExit = start;
+            int totalNodesToDo = (tiles.Rows - 1) * (tiles.Columns - 1) / 4;
+            int[] keyNodes = new int[numKeys];
+            for (int i = 0; i < numKeys; i++)
+            {
+                do
+                {
+                    keyNodes[i] = (random.Next(totalNodesToDo) + random.Next(totalNodesToDo)) / 2; //try to get it halfway
+                } while (isDuplicate(keyNodes, i));
```

Comment:
I looked at the isDuplicate method, it seems to me you should be able to do this using somethign like this

`keyNodes.GetRange(0, i).Contains(keyNodes[i])`